monkey/infection_monkey/master/ip_scanner.py

```diff
@@ -65,6 +69,10 @@ def _scan_addresses(
                 logger.info(f"Scanning {address.ip}")
 
                 ping_scan_data = self._puppet.ping(address.ip, icmp_timeout)
+                self._agent_event_queue.publish(
+                    PingScanEvent(source=get_agent_id(), scan_data=ping_scan_data)
```

Comment:
I created a branch for some of the island-side stuff. The Island needs to know which machines were involved in the ping. It should be able to get the first one using the agent ID. If you set the `target` to be the `IPv4Address` of the machine we're pinging, then that would be sufficient